[mbed-os rtos]: set_stack_heap() calculates wrong heap size when heap size is low

### Description
- Type: Bug 

---------------------------------------------------------------
## Bug

**Target**
all

**Toolchain:**
all

**meed-os sha:**
actual master

In set_stack_heap(void) the mbed_heap start/size is calculated:

```    
    uint32_t free_size = HEAP_SIZE;
...
free_size -= mbed_stack_isr_size;
#endif

    /* Heap - everything else */
    mbed_heap_size = free_size;
    mbed_heap_start = free_start;
```

so when the HEAP_SIZE is low (due to many static vars in .bss) and less than mbed_stack_isr_size, the free_size becomes negative and the uint overflows. That gives you terrific 4GB of memory :-)

that gives the following error:
./mbed-os/rtos/rtx/TARGET_CORTEX_M/RTX_CM_lib.h:484:24: error: expression in static assertion is not constant

But the MBED_ASSERT compiles without errors:
```
void set_stack_heap(void) {

    unsigned char *free_start = HEAP_START;
    uint32_t free_size = HEAP_SIZE;

#ifdef ISR_STACK_START
    /* Interrupt stack explicitly specified */
    mbed_stack_isr_size = ISR_STACK_SIZE;
    mbed_stack_isr_start = ISR_STACK_START;
#else
    /* Interrupt stack -  reserve space at the end of the free block 		*/
    /* if the heap size is too low, it's a severe fault				 		*/
    /* this means not enough RAM for rtos or too many variables in data+bss */
    MBED_ASSERT(HEAP_SIZE >= ISR_STACK_SIZE);

    mbed_stack_isr_size = ISR_STACK_SIZE;
    mbed_stack_isr_start = free_start + free_size - mbed_stack_isr_size;
    free_size -= mbed_stack_isr_size;
#endif
```

--- a/rtos/rtx/TARGET_CORTEX_M/RTX_CM_lib.h
+++ b/rtos/rtx/TARGET_CORTEX_M/RTX_CM_lib.h
@@ -478,7 +478,7 @@ void set_stack_heap(void) {
     mbed_stack_isr_start = ISR_STACK_START;
 #else
     /* Interrupt stack -  reserve space at the end of the free block */
-    mbed_stack_isr_size = ISR_STACK_SIZE;
+    mbed_stack_isr_size = ISR_STACK_SIZE < free_size ? ISR_STACK_SIZE : free_size;
     mbed_stack_isr_start = free_start + free_size - mbed_stack_isr_size;
     free_size -= mbed_stack_isr_size;
 #endif

